flocker/node/agents/test/blockdevicefactory.py

```diff
@@ -179,7 +178,9 @@ def _openstack(**config):
     # case) instead of forcing me to figure out how to upper case things in
     # bash (I already learned a piece of shell syntax today, once is all I can
     # take).
-    region = environ['FLOCKER_FUNCTIONAL_TEST_OPENSTACK_REGION'].upper()
+    region = environ.get('FLOCKER_FUNCTIONAL_TEST_OPENSTACK_REGION')
```

Comment:
There's a bug in the Keystone Server / client API in our PistonCloud test environment which means that it doesn't respond well to region slugs.
This allows me to run the tests in the PistonCloud by omitting the region slug from the environment.